lib/src/id_prefix.rs

```diff
@@ -92,12 +97,19 @@ impl IdIndexSourceEntry<ChangeId> for &'_ (CommitId, ChangeId) {
     }
 }
 
-#[derive(Default)]
 pub struct IdPrefixContext {
     disambiguation: Option<DisambiguationData>,
+    extensions: Vec<Arc<dyn SymbolResolverExtension>>,
```

Comment:
Since `IdPrefixContext` needs `extensions` to evaluate a disambiguation revset, it might be better to pass `Arc<RevsetExpressions>` instead.

That will probably allow us to remove `Arc` from `RevsetExtensions { symbol_resolvers: Vec<Arc<dyn SymbolResolverExtension>> }`.